pkg/tools/spiffeutils/grpcoptions.go

```diff
@@ -67,5 +68,6 @@ func SpiffeCreds(peer TLSPeer, timeout time.Duration) grpc.ServerOption {
 	if err != nil {
 		return grpc.EmptyServerOption{}
 	}
+	tlsConfig.InsecureSkipVerify = true
```

Comment:
As I understand InsecureSkipVerify=true should be used only in testing.
Do we plan to use `SpiffeCreds` only in testing?
If not then I think we can add options to this method something like `WithInsecure()`

Reply:
Typically for most TLS use cases require it when connecting from a client. The hostname is verified against the certificate. However, this ends up failing in SPIFFE. The official SPIFFE golang repo also sets this skip for the same reason.

Reply:
ok, got it.